be/src/vec/functions/round.h

```diff
@@ -696,49 +679,236 @@ class FunctionRounding : public IFunction {
         return Status::OK();
     }
 
-    ColumnNumbers get_arguments_that_are_always_constant() const override { return {1}; }
+    ColumnNumbers get_arguments_that_are_always_constant() const override { return {}; }
+    // SELECT number, func(123.345, 1) FROM numbers("number"="10")
+    // should NOT behave like two column arguments, so we can not use const column default implementation
+    bool use_default_implementation_for_constants() const override { return false; }
```

Comment:
which case should not use default impl of constant? seems wired?

Reply:
This part tries to keep the same with truncate's behavior from [PR#32746.](https://github.com/apache/doris/pull/32746) Specifically, it said that:
> Function truncate can use column as its scale argument. The precision infer rule of truncate is same with mysql:
> 
> When using literal as scale, result scale is dependent on the integer literal;
> When using column as scale, result will always have same scale with input decimal.

Reply:
OK, I will optimize.

Reply:
@HappenLee  @superdiaodiao for query like `SELECT number, func(123.345, 1) FROM numbers("number"="10")`, we should not use default impl for const. If we do, the result will be like `123.300`, but actually, we need `123.3` as result.